apputils/config_manager.py

```diff
@@ -0,0 +1,83 @@
+"""Configuration Manager."""
+from celery.schedules import crontab
+from config_validator import ConfigValidator
+from string import Template
+import ConfigParser
+import os
+
+
+class ConfigManager(object):
+    def __init__(self, config_path):
+        self.scheduled_tasks = {}
+        self.load_config_files(config_path)
+
+    def load_config_files(self, config_path):
+        """Load all valid config files from path, print task config to stdout.
+
+        Args:
+            config_path(str): Path to configuration directory.
+
+        """
+        files_in_dir = [os.path.join(config_path, f)
+                        for f in os.listdir(config_path)
+                        if os.path.isfile(os.path.join(config_path, f))]
+        for target in sorted(files_in_dir):
+            print("ConfigManager: Parsing config file: %s" % target)
+            config = self.get_scheduled_task_config_from_file(target)
+            if ConfigValidator.validate_config(config):
+                task_name = config["task_config"]["task_name"]
+                self.scheduled_tasks[task_name] = config
+                # Coerce `read_only` to boolean
+                if self.scheduled_tasks[task_name]["task_config"]["read_only"] == "true":  # NOQA
+                    self.scheduled_tasks[task_name]["task_config"]["read_only"] = True  # NOQA
+                else:
+                    self.scheduled_tasks[task_name]["task_config"]["read_only"] = False  # NOQA
+        self.print_tasks(self.scheduled_tasks)
+
+    def beat_tasks_from_config(self):
+        """Return dictionary that describes celerybeat tasks, from config."""
+        beats = {}
+        for j in self.scheduled_tasks.items():
+            beats[j[0]] = {'task': 'halocelery.tasks.generic_bound_containerized_task',  # NOQA
+                           'schedule': crontab(hour=j[1]["schedule"]["hour"],
+                                               minute=j[1]["schedule"]["minute"],  # NOQA
+                                               day_of_week=j[1]["schedule"]["day_of_week"],  # NOQA
+                                               day_of_month=j[1]["schedule"]["day_of_month"],  # NOQA
+                                               month_of_year=j[1]["schedule"]["month_of_year"]),  # NOQA
+                           'args': (j[1]["task_config"]["image"],
+                                    j[1]["env_literal"],
+                                    j[1]["env_expand"],
+                                    j[1]["task_config"]["retry"],
+                                    j[1]["log_config"],
+                                    j[1]["task_config"]["read_only"])}
+        return beats
+
+    @classmethod
+    def format_task(cls, task):
```

Comment:
Can all functions that specifically format and prints stuff be moved to a separate class? It's confusing to have formatting functions scattered through each file. should be consolidated.

Reply:
Done.